packages/esm-patient-appointments-app/src/hooks/useAppointmentService.tsx

```diff
@@ -0,0 +1,31 @@
+import useSWR from 'swr';
+import { useMemo } from 'react';
+import { Service } from '../types';
+import { openmrsFetch } from '@openmrs/esm-framework';
+
+type status = 'pending' | 'resolved' | 'error';
+
+const useAppointmentService = () => {
+  const { data = { data: [] }, error } = useSWR<{ data: Array<Service> }, Error>(
+    '/ws/rest/v1/appointmentService/all/full',
+    openmrsFetch,
+  );
+
+  const status = useMemo<status>(() => {
+    if (!error && !data) {
+      return 'pending';
+    }
+    if (error) {
+      return 'error';
+    }
+    if (data) {
+      return 'resolved';
+    }
+  }, [data, error]);
+
+  return useMemo(() => {
+    return { status: status, services: data?.data, error };
+  }, [error, data, status]);
+};
```

Comment:
If we decide to keep this change you can slim this down even further:

```suggestion
const useAppointmentService = () => {
  return useSWR<{ data: Array<Service> }, Error>(
    '/ws/rest/v1/appointmentService/all/full',
    openmrsFetch,
  );
};
```

This can be used like this (i.e. there is no need for keeping a custom status around as it can easily be "calculated" from the SWR result):

```ts
const { data, error } = useAppointmentService();

// status === 'pending' can for example be replaced with this:
const isPending = !error && !data;
```